jaxws-ri/tools/wscompile-jdk9/src/main/java/com/sun/tools/ws/ant/WsGen.java

```diff
@@ -66,20 +68,22 @@ protected void setupForkCommand(String className) {
 
         getCommandline().createClasspath(getProject()).append(new Path(getProject(), antcp));
 
-        boolean addModules = true;
+        boolean addModules = ModuleFinder.ofSystem().find("java.xml.ws").isPresent();
```

Comment:
Maybe we should not add the module if the java.xml.ws classes are already available, for example, if Class.forName(javax.xml.ws.Endpoint) is available.

Reply:
ModuleFinder.ofSystem().find("java.xml.ws") will check if this module is in the run-time image which is not the same thing as checking if the module is in the boot layer. To check if the module is in the boot layer you can use ModuleLayer.boot().findModule("java.xml.ws").

Reply:
Class.forName(javax.xml.ws.Endpoint) will be available even if put on CP, so it doesn't look to work.
I will pick Alan's suggestion: ModuleFinder.ofSystem().find("java.xml.ws")

Reply:
Looking into it `ModuleLayer.boot().findModule("java.xml.ws")` is also not what I want. This would check if module has been already added to boot layer module graph wherever from runtime image or standalone jar. If it is there already there is no need to add it. Thinking more about it it may be best to remove `--add-modules` code completely.